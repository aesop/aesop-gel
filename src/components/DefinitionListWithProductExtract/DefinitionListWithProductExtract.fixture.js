export default {
  dataTestRef: 'test_id',
  product: {
    image: {
      altText: 'Product Extract',
      small:
        'https://www.aesop.com/medias/Aesop-Hand-Resurrection-Rinse-Free-Hand-Wash-500mL-large.png?context=bWFzdGVyfGltYWdlc3wxNjEyNDF8aW1hZ2UvcG5nfGltYWdlcy9oNTgvaDRkLzkwMTkyNTQ5OTcwMjIucG5nfGU5YTRhMDk0Y2FjZWVmYzMyN2MxYTEwZTA2NjJhNTY1MDAwZmRkNzBmOWQ2MGIzYTNiYWM0MjVmODFlOGFmYzc',
    },
    name: 'The sea',
    url: '/',
  },
  works: 'Works well with',
  items: [
    {
      description: 'Proin vulputate eros quis lectus pretium gravida.',
      id: '1',
      term: 'Aliquam',
    },
    {
      description: 'Pulvinar nulla enim, eu elementum lacus vehicula sed.',
      id: '2',
      term: 'Etiam eu augue quam',
    },
    {
      description: 'Sed ac ipsum vel risus aliquam rutrum.',
      id: '3',
      term: 'Duis at elit',
    },
  ],
};
