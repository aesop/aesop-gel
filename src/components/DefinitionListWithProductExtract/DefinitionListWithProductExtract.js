import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import DefinitionList from '../DefinitionList/DefinitionList';
import ProductExtract from '../ProductExtract/ProductExtract';

const DefinitionListWithProductExtract = ({
  dataTestRef,
  isVisible,
  itemBottomBorder,
  items,
  product,
  productBottomBorder,
  productItemNumber,
  theme,
  works,
}) => {
  return (
    <Fragment>
      <DefinitionList
        hasBottomBorder={itemBottomBorder}
        isVisible={isVisible}
        items={items}
        theme={theme}
      />
      <ProductExtract
        dataTestRef={dataTestRef}
        hasBottomBorder={productBottomBorder}
        isVisible={isVisible}
        itemNum={productItemNumber}
        product={product}
        theme={theme}
        works={works}
      />
    </Fragment>
  );
};

DefinitionListWithProductExtract.propTypes = {
  className: PropTypes.string,
  dataTestRef: PropTypes.string.isRequired,
  isVisible: PropTypes.bool,
  itemBottomBorder: PropTypes.bool,
  items: PropTypes.array,
  productBottomBorder: PropTypes.bool,
  productItemNumber: PropTypes.number,
  product: PropTypes.shape({
    image: PropTypes.object,
    name: PropTypes.string,
    url: PropTypes.string,
  }),
  theme: PropTypes.oneOf(['dark', 'light']),
  works: PropTypes.string,
};

DefinitionListWithProductExtract.defaultProps = {
  className: undefined,
  dataTestRef: undefined,
  isVisible: true,
  itemBottomBorder: false,
  items: [],
  productBottomBorder: false,
  productItemNumber: 1,
  product: {
    image: {
      altText: 'Product Extract',
    },
  },
  theme: 'dark',
  works: 'Works well with',
};

export default DefinitionListWithProductExtract;
