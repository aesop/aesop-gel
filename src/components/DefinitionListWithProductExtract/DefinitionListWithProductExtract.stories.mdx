import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import DefinitionListWithProductExtract from './DefinitionListWithProductExtract';
import DefinitionListWithProductExtractFixture from './DefinitionListWithProductExtract.fixture';

<Meta
  title="Components/DefinitionListWithProductExtract"
  component={DefinitionListWithProductExtract}
/>

# DefinitionListWithProductExtract

<Preview>
  <Story name="Base component">
    <DefinitionListWithProductExtract
      dataTestRef={DefinitionListWithProductExtractFixture.dataTestRef}
      product={DefinitionListWithProductExtractFixture.product}
      productBottomBorder={knobs.boolean(
        'productBottomBorder',
        false,
        'Presentation',
      )}
      itemBottomBorder={knobs.boolean(
        'itemBottomBorder',
        false,
        'Presentation',
      )}
      productItemNumber={knobs.select(
        'itemNum',
        [1, 2, 3, 4],
        4,
        'Presentation',
      )}
      isVisible={knobs.boolean('isActive', true, 'Presentation')}
      items={DefinitionListWithProductExtractFixture.items}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
      works={DefinitionListWithProductExtractFixture.works}
    />
  </Story>
</Preview>

## API

<Props of={DefinitionListWithProductExtract} />
