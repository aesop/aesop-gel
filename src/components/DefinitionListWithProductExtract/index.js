/* istanbul ignore file */
import { DefinitionListWithProductExtract } from './DefinitionListWithProductExtract';

export { DefinitionListWithProductExtract };
