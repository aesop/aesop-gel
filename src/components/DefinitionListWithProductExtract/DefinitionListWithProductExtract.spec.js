import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import DefinitionListWithProductExtract from './DefinitionListWithProductExtract';

configure({ adapter: new Adapter() });

describe('<DefinitionListWithProductExtract />', () => {
  it('should be defined', () => {
    expect(DefinitionListWithProductExtract).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer.create(<DefinitionListWithProductExtract />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
