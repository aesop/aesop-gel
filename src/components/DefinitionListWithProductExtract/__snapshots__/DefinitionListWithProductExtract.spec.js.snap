// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefinitionListWithProductExtract /> renders base component correctly 1`] = `
<div
  className="base dark slideIn item1"
>
  <div
    className="test delayAnimation slideIn"
  >
    <div>
      <h4
        className="base xXSmall dark heading delayAnimation slideIn"
      >
        <span>
          Works well with
        </span>
      </h4>
      <div
        className="productName delayAnimation slideIn"
      >
        <div>
          <span />
        </div>
      </div>
    </div>
    <div
      className="productImage delayAnimation slideIn"
    >
      <picture
        className="base picture dark"
      >
        <img
          alt="Product Extract"
        />
      </picture>
    </div>
  </div>
</div>
`;
